tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -395,3 +395,172 @@ def test_invalid(self):
         ng_a = xp.ones((self.n, self.n), dtype='i')
         with pytest.raises(TypeError):
             sp.linalg.cg(ng_a, b, atol=self.atol)
+
+
+@testing.parameterize(*testing.product({
+    'x0': [None, 'ones'],
+    'M': [None, 'jacobi'],
+    'atol': [None, 'select-by-dtype'],
+    'b_ndim': [1, 2],
+    'restart': [None, 10],
+}))
+@unittest.skipUnless(scipy_available, 'requires scipy')
+@testing.gpu
+class TestGmres(unittest.TestCase):
+    n = 30
+    density = 0.2
+    _atol = {'f': 1e-5, 'd': 1e-12}
+
+    def _make_matrix(self, dtype, xp):
+        dtype = numpy.dtype(dtype)
+        shape = (self.n, self.n)
+        a = testing.shaped_random(shape, xp, dtype=dtype.char.lower(), scale=1)
+        if dtype.char in 'FD':
+            a = a + 1j * testing.shaped_random(
+                shape, xp, dtype=dtype.char.lower(), scale=1)
```

Comment:
Any difference from passing a complex dtype to `testing.shaped_random`'s `dtype` parameter?

Reply:
Probably, no particular difference. I just copied the one used in TestCg. Will fix it in the suggested way.